### PR TITLE
[MISC] deprecate seqan3::views::to_upper

### PR DESCRIPTION
**Pull request by MitraDarja**
Resolves https://github.com/seqan/product_backlog/issues/334

### CODE REVIEW

File: include/seqan3/range/views/to_upper.hpp [hunk 1]
```diff
@@ -7,70 +7,10 @@
...
-    static_assert(builtin_character<std::remove_cvref_t<decltype(in)>>,
-                  "The value type of seqan3::views::to_upper must model the seqan3::builtin_character.");
-    return seqan3::to_upper(in);
-})};
-
-//!\}
-
-} // namespace seqan3::views
+SEQAN3_DEPRECATED_HEADER(
+   "This header is deprecated and will be removed in SeqAn-3.1.0.")
```
Comment (marehr) [inline include/seqan3/range/views/to_upper.hpp:16]: memo to myself: this is an interesting idea :)

File: include/seqan3/range/views/to_upper.hpp [hunk 2]
```diff
@@ -61,15 +61,14 @@ namespace seqan3::views
...
+ * \deprecated Use std::views::transform([](auto && chr){return std::toupper(chr)});
  */
-inline auto const to_upper = deep{std::views::transform([] (auto const in) noexcept
-{
-    static_assert(builtin_character<std::remove_cvref_t<decltype(in)>>,
-                  "The value type of seqan3::views::to_upper must model the seqan3::builtin_character.");
-    return seqan3::to_upper(in);
-})};
+ #ifdef SEQAN3_DEPRECATED_310
+ SEQAN3_DEPRECATED_310 inline constexpr auto drop = detail::drop_fn{};
```
Comment (marehr) [inline include/seqan3/range/views/to_upper.hpp:70]: this should be to_upper defintion as before :)

File: include/seqan3/range/views/to_upper.hpp [hunk 3]
```diff
@@ -63,13 +63,17 @@ namespace seqan3::views
...
+ *
+ * \deprecated Use std::views::transform([](auto && chr){return std::toupper(chr)});
  */
-inline auto const to_upper = deep{std::views::transform([] (auto const in) noexcept
+#ifdef SEQAN3_DEPRECATED_310
+SEQAN3_DEPRECATED_310 inline auto const to_upper = deep{std::views::transform([] (auto const in) noexcept
 {
     static_assert(builtin_character<std::remove_cvref_t<decltype(in)>>,
-                  "The value type of seqan3::views::to_upper must model the seqan3::builtin_character.");
+                 "The value type of seqan3::views::to_upper must model the seqan3::builtin_character.");
```
Comment (Irallia) [inline include/seqan3/range/views/to_upper.hpp:73]: This looks like a miscorrection.

**Review by Irallia — APPROVED** [review]
LGFM, found just a false correction. 💅

File: include/seqan3/range/views/to_upper.hpp (hunk 3 above)
Comment (marehr) [inline include/seqan3/range/views/to_upper.hpp:73]:
```suggestion
                  "The value type of seqan3::views::to_upper must model the seqan3::builtin_character.");
```

**Review by marehr — APPROVED** [review]

**Pull request merged**